chi_edge/cli.py

```diff
@@ -11,162 +11,250 @@
-        if sudo_password:
-            host_vars.setdefault("ansible_sudo_pass", sudo_password)
-
-    if device_type == "nano":
-        # Jetson Nanos use L4T distribution based on Ubuntu 18.04, which
-        # doesn't have docker-ce built for stable.
-        host_vars.setdefault("docker_package", "docker")
-
-    dirname = os.path.dirname(__file__)
-    filename = os.path.join(dirname, "ansible/defaults.yml")
-    with open(filename, "r") as f:
-        defaults = yaml.safe_load(f)
-        for key, value in defaults.items():
-            host_vars.setdefault(key, value)
-
-    return ansible.run(
-        "setup_edge_dev.yml", host, group=device_type, host_vars=host_vars
-    )
-
-
-@cli.command(help="Bake a balena image")
+    """Register a device on the CHI@Edge testbed as DEVICE_NAME.
+
+    This is the first step to enrolling an edge device into the testbed and will result
+    in a configuration record for your device being added to the testbed registry. The
+    testbed will generate per-device configuration as a result of this registration,
+    which you can then use to "bake" a generic base OS image for your target device
+    (see: bake command.)
+
+    As a result of registration, your device will be issued a UUID, which you can use
+    to interact with the testbed via other SDK methods.
+
+    \b
+    Supported devices
+    -----------------
+
+    Currently a limited number of machine names (types/classes of devices) are supported:
+
+    \b
+      raspberrypi3-64: Raspberry Pi 3 (using 64bit OS)
+      raspberrypi4-64: Raspberry Pi 4 (using 64bit OS)
+
+    \b
+    Naming your device
+    ------------------
+
+    Every device registered on the testbed requires a machine-friendly name. This must
+    be unique on the testbed; for help choosing a name with maximum utility, please
+    refer to the enrollment documentation:
+    https://chameleoncloud.gitbook.io/chi-edge/device-enrollment/edge-sdk
+
+    """
+    with doni_error_handler("failed to register device"):
+        device = (
+            doni_client()
+            .post(
+                "/v1/hardware/",
+                json={
+                    "name": device_name,
+                    "hardware_type": "device.balena",
+                    "properties": {
+                        "application_credential_id": application_credential_id,
+                        "application_credential_secret": application_credential_secret,
+                        "contact_email": contact_email,
+                        "machine_name": machine_name,
+                    },
+                },
+            )
+            .json()
+        )
+        print_device(device)
+
+
+@device.command(cls=BaseCommand, short_help="list registered devices")
+def list():
+    with doni_error_handler("failed to list devices"):
+        devices = doni_client().get("/v1/hardware/").json()["hardware"]
+        table = make_table()
+        table.add_column("Name")
```

Comment:
this is very neat :) it would be cool to add it to our other CLI tools

Reply:
Yeah, I was pretty happy w/ it overall. It doesn't pair very easily with also supporting generic output but I guess that's OK, it's pretty easy to just passthrough the JSON responses.